projects/services/consensus-service-typescript/src/transaction-pool-service.ts

```diff
@@ -8,17 +8,29 @@ import { TransactionPool } from "orbs-core-library";
 
 export default class TransactionPoolService extends Service {
   private transactionPool: TransactionPool;
+  private pollInterval: NodeJS.Timer;
+  private pollIntervalMs = 2000;
 
   public constructor(transactionPool: TransactionPool, serviceConfig: ServiceConfig) {
     super(serviceConfig);
     this.transactionPool = transactionPool;
   }
 
   async initialize() {
+    this.pollInterval = setInterval(() => {
```

Comment:
It's a bit hacky but we can wait with a better solution for the service management layer. For now:
1. Please make it more maintainable and readable by moving this logic to separate functions (startPoolSizeMonitor() or something of that sort).
2. For the sake of memory optimization, you better export a new getPoolSize() method that calculates the size of the pool from the raw queue instead of fetching all transactions

Reply:
Done.